### PR TITLE
789 ens support

### PR DESCRIPTION
**Pull request by tiagofilipenunes**
#789

- React Queries
	- [x] addressToENS
	- [x] ENSToAddress
- Update components
	- [x] Update RightMenuWallet
	- [x] Explorer: validate valid input (ENS or address)
	- [x] Explorer: search input field keeps ENS input
- Others
	- [x] Check address is valid in addressToENS query
	- [x] Check ENS is valid in ENSToAddress query
	- [x] Support emojis for ENS domains (as well as emojis with longer sizes)
	- [x] Issue with explorer/wallet/{ens.eth} slug
		- explorer/wallet/address.eth doesn't refresh but explorer/wallet/address.eth/ works
		- Related issue: https://github.com/vitejs/vite/issues/2245

As part of the [Frontend guidelines from ENS](https://docs.ens.domains/dapp-developer-guide/front-end-design-guidelines)
- Should use iconify or blockies to identify address if it is not shown fully to the user
- Should add a tooltip or in the wallet modal show the full address to check the user is connected with the correct wallet
- Should trim ENS name differently than address (trim from the end)

### CODE REVIEW

**Comment by tiagofilipenunes** [comment]
# Useful addresses to test
0xkeyrock.eth = 0x1855f41B8A86e701E33199DE7C25d3e3830698ba
binance (no ENS) = 0x4a3ca0cb4dcf6bc573ca4b538cab5fb0e01b5f19
bancorians.eth = 0x6439abb662e3E5F0eb95DcD25A576A722383d8d2
bancoralchemist.eth = 0x21dBF2e0a5f377F7098439B64fb5A67a5d18AF45
😭😭😭.eth = 0xD75337ceF33ace428a18533A1fb791aCB1Eff239

File: package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "ua-parser-js": "^1.0.35",
     "use-async-effect": "^2.2.7",
     "vite": "^4.4.9",
+    "vite-plugin-rewrite-all": "^1.0.1",
```
Comment (zavelevsky) [inline package.json:90]: why do you need this?

File: src/components/core/menu/mainMenu/MainMenuRightWallet.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ import { FC, useMemo } from 'react';
 import { carbonEvents } from 'services/events';
 import { useStore } from 'store';
 import { shortenString } from 'utils/helpers';
+import { useGetEnsFromAddress } from '../../../../libs/queries/chain/ens';
```
Comment (zavelevsky) [inline src/components/core/menu/mainMenu/MainMenuRightWallet.tsx:18]: why use a relative path? We avoid those.

**Comment by ashachaf** [comment]
- [ ] ens address validation
- open explorer
- type any string `sdkjfhskjhfs`
- see invalid wallet error message
- add `.[any letter]`
expected: see the error if the ens name is invalid (doesnt exist or doesnt represent a wallet)
actual: there is no error message.

**Comment by tiagofilipenunes** [comment]
> add .[any letter]
expected: see the error if the ens name is invalid (doesnt exist or doesnt represent a wallet)
actual: there is no error message.

As per the [ENS frontend guidelines](https://docs.ens.domains/dapp-developer-guide/front-end-design-guidelines), ENS domains can be .xyz, .eth, .luxe, .kred, .art, .club, etc:


> Not all ENS names end with .eth: ENS names normally end with .eth. However other top-level domains (TLD) are currently supported (.xyz, .luxe, .kred, .art, .club) and more will be in the future. Consider this if you are thinking about displaying the TLD part in the truncated view of long names.

Can change the regular expression check for only the domains from the list above, but it might restrict future ENS name endings.

**Review by zavelevsky — APPROVED** [review]
Code LGTM - but please make sure you "audit" every new dependency you add (graphemer) - that it's not on any risk list and that it's maintained

**Comment by tiagofilipenunes** [comment]
> Code LGTM - but please make sure you "audit" every new dependency you add (graphemer) - that it's not on any risk list and that it's maintained

Thanks, will do!

**Pull request merged**